File: app/schemas/scenario.py
from datetime import date, datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field

class ScenarioIn(BaseModel):
    # פרמטרים עיקריים לתרחיש חישוב
    planned_termination_date: Optional[date] = None
    retirement_age: Optional[int] = Field(None, ge=50, le=90)
    monthly_expenses: Optional[float] = Field(None, ge=0)
    other_incomes_monthly: Optional[float] = Field(None, ge=0)

class CashflowPoint(BaseModel):
    date: date
    inflow: float
    outflow: float
    net: float

class ScenarioOut(BaseModel):
    # תקציר תוצאות
    seniority_years: float
    grant_gross: float
    grant_exempt: float
    grant_tax: float
    grant_net: float
    pension_monthly: float
    indexation_factor: float
    cashflow: List[CashflowPoint]

class ScenarioCreateIn(BaseModel):
    # יצירת תרחיש חדש עם שם
    scenario_name: str = Field(..., min_length=1, max_length=255)
    planned_termination_date: Optional[date] = None
    monthly_expenses: Optional[float] = Field(None, ge=0)
    # Planning flags
    apply_tax_planning: bool = False
    apply_capitalization: bool = False
    apply_exemption_shield: bool = False
    # Additional parameters
    other_parameters: Dict[str, Any] = Field(default_factory=dict)

class ScenarioCreateResponse(BaseModel):
    # תגובה פשוטה ליצירת תרחיש
    scenario_id: int

class ScenarioCreateOut(BaseModel):
    # תגובה ליצירת תרחיש + תוצאות חישוב
    scenario_id: int
    seniority_years: float
    grant_gross: float
    grant_exempt: float
    grant_tax: float
    grant_net: float
    pension_monthly: float
    indexation_factor: float
    cashflow: List[CashflowPoint]

class ScenarioListItem(BaseModel):
    # פריט ברשימת תרחישים
    id: int
    scenario_name: str
    apply_tax_planning: bool
    apply_capitalization: bool
    apply_exemption_shield: bool
    created_at: datetime

class ScenarioListOut(BaseModel):
    # רשימת תרחישים
    scenarios: List[ScenarioListItem]

File: frontend/src/types/scenario.ts
export interface Scenario {
  id: number;
  scenario_name: string;
  apply_tax_planning: boolean;
  apply_capitalization: boolean;
  apply_exemption_shield: boolean;
  created_at: string;
}

export interface ScenarioResult {
  id: number;
  scenario_name: string;
  client_id: number;
  apply_tax_planning: boolean;
  apply_capitalization: boolean;
  apply_exemption_shield: boolean;
  seniority_years: number;
  grant_gross: number;
  grant_exempt: number;
  grant_tax: number;
  grant_net: number;
  pension_monthly: number;
  indexation_factor: number;
  cashflow: Array<{
    date: string;
    inflow: number;
    outflow: number;
    net: number;
  }>;
  created_at: string;
}

export interface ScenariosProps {
  clientId: number;
}

export interface ScenarioFormData {
  scenario_name: string;
  planned_termination_date: string;
  monthly_expenses: string;
  apply_tax_planning: boolean;
  apply_capitalization: boolean;
  apply_exemption_shield: boolean;
  other_parameters: Record<string, any>;
}

export const INITIAL_FORM_STATE: ScenarioFormData = {
  scenario_name: '',
  planned_termination_date: '',
  monthly_expenses: '',
  apply_tax_planning: false,
  apply_capitalization: false,
  apply_exemption_shield: false,
  other_parameters: {}
};
